src/test/java/com/inventorsoft/junit/mapper/BookMapperTest.java

```diff
@@ -0,0 +1,163 @@
+        book.setDescription(DESCRIPTION);
+        book.setAuthor(author);
+
+        authorResponse.setId(AUTHOR_ID);
+        authorResponse.setName(AUTHOR_NAME);
+        authorResponse.setBirthDate(AUTHOR_BIRTHDAY);
+
+        bookResponse.setId(ID);
+        bookResponse.setTitle(TITLE);
+        bookResponse.setReleaseDate(RELEASE_DATA);
+        bookResponse.setDescription(DESCRIPTION);
+        bookResponse.setAuthor(authorResponse);
+
+        createBookRequest.setTitle(TITLE);
+        createBookRequest.setDescription(DESCRIPTION);
+        createBookRequest.setReleaseDate(RELEASE_DATA);
+        createBookRequest.setAuthorId(AUTHOR_ID);
+
+        updateBookRequest.setTitle("ddad");
+        updateBookRequest.setReleaseDate(LocalDate.of(2033, 2, 2));
+        updateBookRequest.setDescription("aa");
+        updateBookRequest.setAuthorId(4L);
+
+        updateAuthor.setId(4L);
+        updateAuthor.setName("Dmytro Jensen");
+        updateAuthor.setBirthday(LocalDate.of(1970, 2, 2));
+
+        updateBook.setId(1L);
+        updateBook.setTitle("ddad");
+        updateBook.setReleaseDate(LocalDate.of(2033, 2, 2));
+        updateBook.setDescription("aa");
+        updateBook.setAuthor(updateAuthor);
+
+    }
+
+    @Test
+    void mapEntitiesToResponses() {
+        List<Book> books = List.of(book);
+        List<BookResponse> expectedResults = List.of(bookResponse);
+
+        when(authorMapper.mapEntityToResponse(book.getAuthor())).thenReturn(authorResponse);
+
+        List<BookResponse> givenResults = bookMapper.mapEntitiesToResponses(books);
+        System.out.println(givenResults);
+
+        // then
+        assertEquals(expectedResults.size(), givenResults.size());
+        assertArrayEquals(expectedResults.toArray(), givenResults.toArray());
+    }
+
+    @Test
+    void mapEntityToResponse() {
+        when(authorMapper.mapEntityToResponse(book.getAuthor())).thenReturn(authorResponse);
+
+        BookResponse givenResult = bookMapper.mapEntityToResponse(book);
+
+        assertEquals(bookResponse.getId(), givenResult.getId());
+        assertEquals(bookResponse.getTitle(), givenResult.getTitle());
+        assertEquals(bookResponse.getReleaseDate(), givenResult.getReleaseDate());
+        assertEquals(bookResponse.getDescription(), givenResult.getDescription());
+        assertEquals(bookResponse.getAuthor(), givenResult.getAuthor());
+    }
+
+    @Test
+    void mapCreateRequestToEntity() {
+
+        when(authorRepository.findById(AUTHOR_ID)).thenReturn(Optional.of(author));
+        Book givenResult = bookMapper.mapCreateRequestToEntity(createBookRequest);
+
+
+        assertEquals(book.getTitle(), givenResult.getTitle());
+        assertEquals(book.getDescription(), givenResult.getDescription());
+        assertEquals(book.getReleaseDate(), givenResult.getReleaseDate());
+        assertEquals(book.getAuthor(), givenResult.getAuthor());
+    }
+
+    @Test
+    void ifIdAuthorIsNullAnExceptionShouldBeThrown() {
```

Comment:
this test name does not show which method it tests, for example, `findByShouldThrownExeptionWhenNotFound`,
and this test does not test anything, if you run it alone it will not add code coverage %. 
you invoke here `authorRepository.findById` as a tested method, but you test `bookMapper`, in that mapper there is no such method, that method is inside `mapCreateRequestToEntity`, so you have to test that condition there..

Create test with name `mapCreateRequestToEntityShouldThrowRuntimeExceptionWhenAuthorIsNotFound`, body should be something like:
```suggestion
      when(authorRepository.findById(AUTHOR_ID)).thenReturn(Optional.empty());
      assertThrows(RuntimeException.class, () -> bookMapper.mapCreateRequestToEntity(createBookRequest));
```